contrib/guix/manifest.scm

```diff
@@ -139,12 +139,17 @@ chain for " target " development."))
     "--enable-default-ssp" "yes")
     "--enable-default-pie" "yes"))
 
+(define building-on (string-append (list-ref (string-split (%current-system) #\-) 0) "-guix-linux-gnu"))
+
+(define (explicit-build-configure package)
+  (package-with-extra-configure-variable package "--build" building-on))
+
 (define* (make-bitcoin-cross-toolchain target
                                        #:key
-                                       (base-gcc-for-libc base-gcc)
+                                       (base-gcc-for-libc (explicit-build-configure base-gcc))
                                        (base-kernel-headers base-linux-kernel-headers)
-                                       (base-libc (make-glibc-with-bind-now (make-glibc-without-werror glibc-2.24)))
-                                       (base-gcc (make-gcc-rpath-link (hardened-gcc base-gcc))))
+                                       (base-libc (explicit-build-configure (make-glibc-with-bind-now (make-glibc-without-werror glibc-2.24))))
+                                       (base-gcc (explicit-build-configure (make-gcc-rpath-link (hardened-gcc base-gcc)))))
```

Comment:
Why configuration of the `base-*` tools being changed?

Should it being changed for `cross-*` tools only?

See [this](https://github.com/hebasto/bitcoin/commits/pr25861/0822.ALT) branch as a working implementation.